openpype/pipeline/workfile/workfile_template_builder.py

```diff
@@ -1825,3 +1832,40 @@ def get_errors(self):
 
     def create_failed(self, creator_data):
         self._failed_created_publish_instances.append(creator_data)
+
+
+def build_first_workfile_from_template_builder():
```

Comment:
This function name doesn't sound right? This returns whether it should build - it doesn't return a "builder"?
Additionally it should allow to take in settings, asset and task name, etc. so that if you want to optimize the call with settings etc that you might've already retrieved, then you can.

```
def should_create_first_workfile(
        project_name=None,
        project_settings=None,
        asset_doc=None,
        asset_name=None,
        task_name=None,
        host_name=Nome
    ):
    """Return whether first workfile should be created for given context"""
```
    project_name = project_name or get_current_project_name()
    if project_settings is None:
        project_settings = get_project_settings(project_name)
     etc.
```

I'd also move the `if build_workfile_profiles["profiles"]:` way higher up before retrieving more context things (directly after getting project settings) so that if no profiles are defined you don't need to do any other querying.